app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,11 @@
+# Controller for user features not provided by Devise - profile pages, etc
+class Users::SessionsController < Devise::SessionsController
+  def new
+    unless Setting.yes? I18n.t( 'allow_user_logins' )
```

Comment:
Consider improving the readability of this line. The combination of the `yes?` and the `I18n.t(...)` feels like it could be clarified with a change like:

```diff
-unless Setting.yes? I18n.t( 'allow_user_logins' )
+unless Setting.allow_user_logins?
```